test/org/jetbrains/plugins/ideavim/extension/highlightedyank/VimHighlightedYankTest.kt

```diff
@@ -0,0 +1,172 @@
+//  fun `test removing previous highlight when entering insert mode`() {
+//    doTest("yyi", code, code, CommandState.Mode.INSERT, CommandState.SubMode.NONE)
+//
+//    assertAllHighlightersCount(0)
+//  }
+
+  fun `test indicating error when incorrect highlight duration was provided by user`() {
+    configureByJavaText(code)
+    typeText(StringHelper.parseKeys(":let g:highlightedyank_highlight_duration = \"500.15\"<CR>"))
+    typeText(StringHelper.parseKeys("yy"))
+
+    Assert.assertEquals(VimPlugin.getMessage(), "highlightedyank: Invalid value of g:highlightedyank_highlight_duration -- For input string: \"500.15\"")
+  }
+  fun `test not indicating error when correct highlight duration was provided by user`() {
+
+    configureByJavaText(code)
+    typeText(StringHelper.parseKeys(":let g:highlightedyank_highlight_duration = \"-1\"<CR>"))
+    typeText(StringHelper.parseKeys("yy"))
+
+    Assert.assertEquals(VimPlugin.getMessage(), "")
+  }
+
+  fun `test indicating error when incorrect highlight color was provided by user`() {
+    configureByJavaText(code)
+
+    listOf("rgba(1,2,3)", "rgba(1, 2, 3, 0.1)", "rgb(1,2,3)", "rgba(260, 2, 5, 6)").forEach { color ->
+      typeText(StringHelper.parseKeys(":let g:highlightedyank_highlight_color = \"$color\"<CR>"))
+      typeText(StringHelper.parseKeys("yy"))
+
+      Assert.assertTrue(color, VimPlugin.getMessage().contains("highlightedyank: Invalid value of g:highlightedyank_highlight_color"))
+    }
+  }
+
+  fun `test indicating error when correct highlight color was provided by user`() {
+    configureByJavaText(code)
+
+    listOf("rgba(1,2,3,5)", "rgba1, 2, 3, 1", "rgba(1, 2, 3, 4").forEach { color ->
+      typeText(StringHelper.parseKeys(":let g:highlightedyank_highlight_color = \"$color\"<CR>"))
+      typeText(StringHelper.parseKeys("yy"))
+
+      Assert.assertEquals("", VimPlugin.getMessage())
+    }
+  }
+
+  @TestWithoutNeovim(reason = SkipNeovimReason.PLUGIN)
+  fun `test highlighting with multiple cursors`() {
+    doTest("yiw", codeWithMultipleCurors, codeWithMultipleCurors, CommandState.Mode.COMMAND, CommandState.SubMode.NONE)
+
+    val highlighters = myFixture.editor.markupModel.allHighlighters
+    assertAllHighlightersCount(3)
+    assertHighlighterRange(12, 15, highlighters[1])
+    assertHighlighterRange(20, 23, highlighters[0])
+    assertHighlighterRange(28, 31, highlighters[2])
+  }
+
+//  @TestWithoutNeovim(reason = SkipNeovimReason.PLUGIN)
+//  fun `test clearing all highlighters with multiple cursors`() {
+//    doTest("yiwi", codeWithMultipleCurors, codeWithMultipleCurors, CommandState.Mode.INSERT, CommandState.SubMode.NONE)
+//
+//    assertAllHighlightersCount(0)
+//  }
+
+  @TestWithoutNeovim(reason = SkipNeovimReason.PLUGIN)
+  fun `test highlighting for a correct default amount of time`() {
+    doTest("yiw", code, code, CommandState.Mode.COMMAND, CommandState.SubMode.NONE)
+
+    assertHappened(DEFAULT_HIGHLIGHT_DURATION.toInt(), 200) {
```

Comment:
I've followed your advice, but the precision has to be quite large :( 200ms is I think the lowest possible amount I've found to be working. 
I've also tried to measure time locally on running IDE, and it works well, the difference between set time and actual highlight duration is max. 5ms

Reply:
I've done some more measurements and I think the reason for this is that the `typeText` takes about 200ms...